compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala

```diff
@@ -440,24 +440,37 @@ case class ApplicationCompiler(typeAliases: Map[String, Type],
       case ExecutableKindWorkflowOutputReorg =>
         // The reorg applet requires higher permissions to organize the output directory.
         Some(DxAccess.empty.copy(project = Some(DxAccessLevel.Contribute)))
+      case ExecutableKindWfInputs(blockPath) =>
+        Some(
+            DxAccess.empty.copy(project = Some(DxAccessLevel.Upload),
+                                allProjects = Some(DxAccessLevel.View))
+        )
+      case ExecutableKindWfOutputs(blockPath) =>
+        Some(DxAccess.empty.copy(allProjects = Some(DxAccessLevel.Upload)))
       case _ =>
         // Scatters need network access, because they spawn subjobs that (may) use dx-docker.
         // We end up allowing all applets to use the network
-        Some(DxAccess.empty.copy(network = Some(Vector("*"))))
+        Some(DxAccess.empty)
```

Comment:
I added two cases for common input applet and output applet to make sure that 

- the input applet can clone files from other accessible projects
https://documentation.dnanexus.com/developer/api/data-containers/cloning#errors
> VIEW access is required in the source container
> UPLOAD access is required in the destination container
- the output applet can upload to all accessible projects.

Reply:
removed

Reply:
pls add tests

Reply:
Good call. Will do. Thanks!